tests/integration/modules/mac_shadow.py

```diff
@@ -0,0 +1,163 @@
+        self.assertTrue(isinstance(ret, dict))
+        self.assertEqual(ret['name'], TEST_USER)
+
+    @destructiveTest
+    def test_get_account_created(self):
+        '''
+        Test shadow.get_account_created
+        '''
+        self.run_function('user.add', [TEST_USER])
+        text_date = self.run_function('shadow.get_account_created', [TEST_USER])
+        obj_date = datetime.datetime.strptime(text_date, '%Y-%m-%d %H:%M:%S')
+        self.assertTrue(isinstance(obj_date, datetime.date))
+
+    @destructiveTest
+    def test_get_last_change(self):
+        '''
+        Test shadow.get_last_change
+        '''
+        self.run_function('user.add', [TEST_USER])
+        text_date = self.run_function('shadow.get_last_change', [TEST_USER])
+        obj_date = datetime.datetime.strptime(text_date, '%Y-%m-%d %H:%M:%S')
+        self.assertTrue(isinstance(obj_date, datetime.date))
+
+    @destructiveTest
+    def test_get_login_failed_last(self):
+        '''
+        Test shadow.get_login_failed_last
+        '''
+        self.run_function('user.add', [TEST_USER])
+        text_date = self.run_function('shadow.get_login_failed_last',
+                                      [TEST_USER])
+        obj_date = datetime.datetime.strptime(text_date, '%Y-%m-%d %H:%M:%S')
+        self.assertTrue(isinstance(obj_date, datetime.date))
+
+    @destructiveTest
+    def test_get_set_maxdays(self):
+        '''
+        Test shadow.get_maxdays
+        Test shadow.set_maxdays
+        '''
+        self.run_function('user.add', [TEST_USER])
+        self.assertTrue(self.run_function('shadow.set_maxdays',
+                                          [TEST_USER, 20]))
+        self.assertEqual(self.run_function('shadow.get_maxdays', [TEST_USER]),
+                         20)
+
+    @destructiveTest
+    def test_get_set_change(self):
+        '''
+        Test shadow.get_change
+        Test shadow.set_change
+        '''
+        self.run_function('user.add', [TEST_USER])
+        self.assertTrue(self.run_function('shadow.set_change',
+                                          [TEST_USER, '02/11/2011']))
+        self.assertEqual(self.run_function('shadow.get_change', [TEST_USER]),
+                         '02/11/2011')
+
+    @destructiveTest
+    def test_get_set_expire(self):
+        '''
+        Test shadow.get_expire
+        Test shadow.set_expire
+        '''
+        self.run_function('user.add', [TEST_USER])
+        self.assertTrue(self.run_function('shadow.set_expire',
+                                          [TEST_USER, '02/11/2011']))
+        self.assertEqual(self.run_function('shadow.get_expire', [TEST_USER]),
+                         '02/11/2011')
+
+    @destructiveTest
+    def test_del_password(self):
+        '''
+        Test shadow.del_password
+        '''
+        self.run_function('user.add', [TEST_USER])
+        self.assertTrue(self.run_function('shadow.del_password', [TEST_USER]))
```

Comment:
The execution module returns true if '*' is found as the password after the password has been deleted. The check is being done there.